cli/azd/pkg/devcenter/platform.go

```diff
@@ -152,19 +152,13 @@ func (p *Platform) ConfigureContainer(container *ioc.NestedContainer) error {
 	})
 
 	// Provision Provider
-	if err := container.RegisterNamedSingleton(string(ProvisionKindDevCenter), NewProvisionProvider); err != nil {
-		return err
-	}
+	container.RegisterNamedScoped(string(ProvisionKindDevCenter), NewProvisionProvider)
```

Comment:
What drives the decision on if something is a singleton vs scoped now?  I feel like any component which consumes a scoped resource should not itself be a singleton?

Reply:
**Singleton**: Only ever create a single instance throughout the lifetime of a root `azd` command
**Scoped**: Create a new instance for each `azd` command.  Any sub comments executed in a workflow will create a new scope
**Transient**: Create a new instance every time i ask for it